### PR TITLE
Cbrune/bfd 189 beneficiary table

### PR DESCRIPTION
**Pull request by cbrunefearless**
### Change Details

Added a check in the rifloader when beneficiary history needs to be updated to check with the old beneficiary against the hicn, hicnunhashed,birthdate, and sex.  The beneficiary id wont change so i didnt compare that field.

### Acceptance Validation

That when a a old history and a new history are compared the right and correct amount of fields are being compared.

### Feedback Requested

Make sure the right fields are compared between beneficiary history entities.
### External References

- BFD-189 (https://jira.cms.gov/browse/BFD-189)

### Security Implications

- [ ] new software dependencies
No

- [ ] altered security controls
 No

- [ ] new data stored or transmitted
new data is being stored in terms of a mbi hash for the beneficiary history table.

mbi hash and not PII/PHI

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

Testing scenarios:
Was able to reproduce the problem in the old code.  It would just persist a new record to the beneficiary history table if they're were any changes.  It was just checking to see if the entity existed before and if it did based on the null check it would create another record.  

I was able to test the change in the code with our integration test and make sure the record if it found the same Birthdate, Hicn, HicnUnhashed, Sex, MbiHash, and medicarebeneficiaryId it will not insert into the beneficiary history table.

### CODE REVIEW

**Comment by jzulim** [comment]
👋 @cbrunefearless - Could you detail the testing scenarios that you've ran through during this development?

i.e. Reproduced problem with `master` code
i.e. With fix in place, loaded duplicate bene records and confirmed no duplicates created

etc.

**Comment by cbrunefearless** [comment]
> 👋 @cbrunefearless - Could you detail the testing scenarios that you've ran through during this development?
> 
> i.e. Reproduced problem with `master` code
> i.e. With fix in place, loaded duplicate bene records and confirmed no duplicates created
> 
> etc.

Added it above

File: apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java [hunk 1]
```diff
@@ -86,4 +89,111 @@ public void computeMbiHash() {
...
+
+    // Undo New beneficiary sex and set it back to old should assert true
+    newBene.setSex(sex);
+    Assert.assertTrue(RifLoader.isBeneficiaryHistoryEqual(newBene, oldBene));
+
+    // New beneficiary mediciarybeneficiaryid is not the same as old should assert false
+    newBene.setMedicareBeneficiaryId(Optional.of("diff"));
+    Assert.assertFalse(RifLoader.isBeneficiaryHistoryEqual(newBene, oldBene));
+
+    // Undo New beneficiary sex and set it back to old should assert true
```
Comment (dtisza1) [inline apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java:162]: comment mismatch?

File: apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java [hunk 2]
```diff
@@ -86,4 +89,111 @@ public void computeMbiHash() {
...
+    newBene.setSex(sex);
+    Assert.assertTrue(RifLoader.isBeneficiaryHistoryEqual(newBene, oldBene));
+
+    // New beneficiary mediciarybeneficiaryid is null and the return result should assert false
+    newBene.setMedicareBeneficiaryId(Optional.empty());
+    Assert.assertFalse(RifLoader.isBeneficiaryHistoryEqual(newBene, oldBene));
+
+    // Undo New beneficiary sex and set it back to old should assert true
```
Comment (dtisza1) [inline apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java:195]: comment mismatch?

File: apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java [hunk 3]
```diff
@@ -86,4 +89,111 @@ public void computeMbiHash() {
...
+    // Check for nulls
+    // New beneficiary hicn unleashed is null and the return result should assert false
+    newBene.setHicnUnhashed(Optional.empty());
+    Assert.assertFalse(RifLoader.isBeneficiaryHistoryEqual(newBene, oldBene));
+
+    // Undo New beneficiary hicn unleashed and set it back to old should assert true
+    newBene.setHicnUnhashed(hicnUnhased);
+    Assert.assertTrue(RifLoader.isBeneficiaryHistoryEqual(newBene, oldBene));
+
+    // New beneficiary hicn unleashed is null and the return result should assert false
```
Comment (dtisza1) [inline apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java:183]: comment mismatch?

Comment (cbrunefearless) [reply apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java:183]: fixed

File: apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java (hunk 1 above)
Comment (cbrunefearless) [reply apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java:162]: fixed

File: apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java [hunk 4]
```diff
@@ -86,4 +89,111 @@ public void computeMbiHash() {
...
+    newBene.setHicnUnhashed(Optional.of("difHicn"));
+    Assert.assertFalse(RifLoader.isBeneficiaryHistoryEqual(newBene, oldBene));
+
+    // Undo New beneficiary hicn unleashed and set it back to old should assert true
+    newBene.setHicnUnhashed(hicnUnhased);
+    Assert.assertTrue(RifLoader.isBeneficiaryHistoryEqual(newBene, oldBene));
+
+    // New beneficiary hicn unleashed is not the same as old should assert false
+    newBene.setSex('F');
+    Assert.assertFalse(RifLoader.isBeneficiaryHistoryEqual(newBene, oldBene));
```
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java:152]: fixed

File: apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java (hunk 2 above)
Comment (cbrunefearless) [reply apps/bfd-pipeline/bfd-pipeline-rif-load/src/test/java/gov/cms/bfd/pipeline/rif/load/RifLoaderTest.java:195]: fixed

**Pull request merged**